### PR TITLE
Fix not picking completed careplan

### PR DESCRIPTION
**Pull request by aurangzaibumer**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes https://github.com/onaio/fhir-resources/issues/2850

**Engineer Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [ ] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/engine/src/main/java/org/smartregister/fhircore/engine/task/FhirCarePlanGenerator.kt [hunk 1]
```diff
@@ -99,6 +100,14 @@ constructor(
         .search<CarePlan> {
           filter(CarePlan.INSTANTIATES_CANONICAL, { value = planDefinition.referenceValue() })
           filter(CarePlan.SUBJECT, { value = subject.referenceValue() })
+          filter(
+            CarePlan.STATUS,
+            { value = of(CarePlanStatus.DRAFT.toCode()) },
+            { value = of(CarePlanStatus.ACTIVE.toCode()) },
+            { value = of(CarePlanStatus.ONHOLD.toCode()) },
+            { value = of(CarePlanStatus.ENTEREDINERROR.toCode()) },
```
Comment (dubdabasoduba) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/task/FhirCarePlanGenerator.kt:108]: @aurangzaibumer I think we can remove this. By the time the CarePlan us marked as entered in error it means the it should be closed

**Review by dubdabasoduba — CHANGES_REQUESTED** [review]

**Comment by ellykits** [comment]
@aurangzaibumer Do we need this PR for the main branch? @dubdabasoduba Can you review this?

**Comment by aurangzaibumer** [comment]
> @aurangzaibumer Do we need this PR for the main branch? @dubdabasoduba Can you review this?

Yes, we do need this PR because app was picking a completed careplan which was a bug. Here's PR to the main https://github.com/opensrp/fhircore/pull/2929

**Comment by ellykits** [comment]
> > @aurangzaibumer Do we need this PR for the main branch? @dubdabasoduba Can you review this?
> 
> Yes, we do need this PR because app was picking a completed careplan which was a bug. Here's PR to the main #2929

Thanks. Kindly create a corresponding PR pointing to `main` branch.

**Comment by aurangzaibumer** [comment]
> > > @aurangzaibumer Do we need this PR for the main branch? @dubdabasoduba Can you review this?
> > 
> > 
> > Yes, we do need this PR because app was picking a completed careplan which was a bug. Here's PR to the main #2929
> 
> Thanks. Kindly create a corresponding PR pointing to `main` branch.

done. should we merge this PR now? or this needs to be closed?

**Review by dubdabasoduba — APPROVED** [review]

**Pull request merged**